File: products/admin-desktop/g5-admin/src/api/client/ssh-profiles.ts
import type { SshProfileAddInput } from "../../types/SshProfileAddInput";
import type { SshProfileDeleteInput } from "../../types/SshProfileDeleteInput";
import type { SshProfileListInput } from "../../types/SshProfileListInput";
import type { SshProfileListResponse } from "../../types/SshProfileListResponse";
import type { SshProfileUpdateInput } from "../../types/SshProfileUpdateInput";
import { invokeCommand } from "./core";

export async function getSshProfileList(
  input: SshProfileListInput,
): Promise<SshProfileListResponse> {
  return invokeCommand<SshProfileListResponse>("cmd_ssh_profile_list", { input });
}

export async function addSshProfile(
  input: SshProfileAddInput,
): Promise<SshProfileListResponse> {
  return invokeCommand<SshProfileListResponse>("cmd_ssh_profile_add", { input });
}

export async function updateSshProfile(
  input: SshProfileUpdateInput,
): Promise<SshProfileListResponse> {
  return invokeCommand<SshProfileListResponse>("cmd_ssh_profile_update", { input });
}

export async function deleteSshProfile(
  input: SshProfileDeleteInput,
): Promise<SshProfileListResponse> {
  return invokeCommand<SshProfileListResponse>("cmd_ssh_profile_delete", { input });
}

File: products/admin-desktop/g5-admin/src-tauri/src/app_state/ssh_profiles.rs
#[cfg(test)]
use super::*;

#[cfg(test)]
#[allow(dead_code)]
impl AppState {
    pub async fn ssh_profile_list(
        &self,
        request_id: &str,
        site_id: &str,
    ) -> Result<SshProfileListResponse, AppError> {
        self.ssh_profile_service().list(request_id, site_id).await
    }

    pub async fn add_ssh_profile(&self, input: SshProfileAddInput) -> Result<(), AppError> {
        self.ssh_profile_service().add(input).await
    }

    pub async fn update_ssh_profile(&self, input: SshProfileUpdateInput) -> Result<(), AppError> {
        self.ssh_profile_service().update(input).await
    }

    pub async fn delete_ssh_profile(&self, input: SshProfileDeleteInput) -> Result<(), AppError> {
        self.ssh_profile_service().delete(input).await
    }
}
